xonsh/environ.py

```diff
@@ -1721,6 +1721,9 @@ def __init__(self, *args, **kwargs):
         self._no_value = object()
         self._orig_env = None
         self._vars = {k: v for k, v in DEFAULT_VARS.items()}
+        self._defaults = {
+            k: v.default for k, v in self._vars.items() if v.default is not None
+        }
```

Comment:
Unfortunately, None is a valid default.  So I think you would nee to use a different sentinal, like `xonsh.tools.DefaultNotGiven`. Also, we have moved everything into a single `_vars` (finally) so we would like to keep it there